homeassistant/components/sensor/imap_email_content.py

```diff
@@ -124,13 +128,15 @@ class EmailContentSensor(Entity):
     """Representation of an EMail sensor."""
 
     def __init__(self, hass, email_reader, name, allowed_senders,
-                 value_template):
+                 value_template, body_regex):
         """Initialize the sensor."""
         self.hass = hass
         self._email_reader = email_reader
         self._name = name
         self._allowed_senders = [sender.upper() for sender in allowed_senders]
         self._value_template = value_template
+        self._body_regex = re.compile(body_regex) if body_regex is not None \
```

Comment:
If the user does specify an invalid regex, this should not simply crash. Either an error should be emitted here, or (even better) validation should take care of this. I don't see a regex validator in `config_validation.py`, but that shouldn't be too hard.